streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1474,7 +1482,7 @@ private void closeToError() {
         if (!setState(State.PENDING_ERROR)) {
             log.info("Skipping shutdown since we are already in " + state());
         } else {
-            final Thread shutdownThread = shutdownHelper(true);
+            final Thread shutdownThread = shutdownHelper(true, Long.MAX_VALUE, false);
```

Comment:
Of all the refactoring, this is probably the "scariest".  @mjsax thoughts?

Reply:
What `timeoutMs`?  

There's no `timeoutMs` in scope.

The method `closeToError` is called a few places; not sure if we could add a timeout to each of them...